### PR TITLE
Refactor/justinyaodu/1290059430 replace admin permission

### PR DESCRIPTION
**Pull request by justinyaodu**
### Administrative Info
Monday Board ID: 1181559666
Make sure your branch name conforms to: `<feature/staging/hotfix/...>/[username]/[Monday Item ID]-[3-4 word description separated by dashes]`. Otherwise, please rename your branch and create a new PR.

### Changes
What changes did you make?
* Renamed the `admin` permission to `user_edit`
* Changed each occurrence of `admin` in the codebase to `user_edit` or `role_management`, as appropriate
* Added test scripts for user edit and role edit routes

### Testing
How did you confirm your changes worked? 
* Logged in as the example president and confirmed that the role settings sidebar option showed up, and that the edit menu showed up on a user's profile
* Logged in as the example developer and confirmed that the above UI elements did not show up
* Ran the role edit test script to confirm that role editing still works
* Ran the user edit test script to confirm that user editing still works

### Confirmation of Change 
Upload a screenshot, if possible. Otherwise, please provide instructions on how to see the change.

| Logged in as President |
| --- |
| screenshot-20210516-135442 |

| Logged in as Developer |
| --- |
| screenshot-20210516-135453 |

### CODE REVIEW

File: backend/services/mock.js [hunk 1]
```diff
@@ -51,7 +51,7 @@ async function createMockData() {
     permissions: {
```
Comment (hahsu) [inline backend/services/mock.js:51]: Can we update these permissions as well, President should have all of the permissions

**Review by hahsu — CHANGES_REQUESTED** [review]
Can we update the PM permissions in mock.js: 
https://github.com/TritonSE/TSE-Oktavian/blob/46d0bb9702fb3b35b75ca56df8816fdab55f5bd4/backend/services/mock.js#L34
PM's should have edit_project permissions.

**Comment by justinyaodu** [comment]
Thanks for the feedback! Let me know if you notice anything else :slightly_smiling_face:

**Review by hahsu — APPROVED** [review]
lgtm, thank you!

**Review by AnsonLee99 — APPROVED** [review]
LGTM!

**Pull request merged**